src/main/scala/shell/microsemi/PolarFireEvalKitShell.scala

```diff
@@ -623,7 +623,7 @@ abstract class PolarFireEvalKitShell(implicit val p: Parameters) extends RawModu
   // UART
   //-----------------------------------------------------------------------
 
-  def connectUART(dut: HasPeripheryUARTModuleImp): Unit = dut.uart.headOption.foreach(connectUART)
+  def connectUART(dut: HasPeripheryUART): Unit = dut.uart.headOption.foreach(connectUART)
```

Comment:
Interesting, seems it has an old bug.

Reply:
No, I am about to open a PR that removes the HasPeripheryXModuleImp pattern out of rocket-chip-blocks

Reply:
Nice.